src/codeu/chat/client/commandline/Chat.java

```diff
@@ -45,11 +49,19 @@ public Chat(Context context) {
   // is willing to take another command, the function will return true. If
   // the system wants to exit, the function will return false.
   //
-  public boolean handleCommand(String line) {
+  public boolean handleCommand(String line) throws IOException {
 
-    final Scanner tokens = new Scanner(line.trim());
-
-    final String command = tokens.hasNext() ? tokens.next() : "";
+    final List<String> args = new ArrayList<>();
+    final Tokenizer tokenizer = new Tokenizer(line);
+    try {
+      for (String token = tokenizer.next(); token != null; token = tokenizer.next()) {
+        args.add(token);
+      }
+    } catch (IOException e) {
+      System.out.println("ERROR: Nothing found in args, IOException thrown. " + e);
```

Comment:
@tsums I know that the IOException occurs because in the next() method, peek() is called within the class and that can thrown an IOException if we've read through all the characters in a current source, but when I wrote this message I was thinking of it like a Scanner where you'd throw the exception because you don't have any input, so there's nothing for next() to read in the Tokenizer class. Not sure if my reasoning is okay, but that's what I was going with.